core/server/worker/src/main/java/alluxio/worker/grpc/DoraWorkerClientServiceHandler.java

```diff
@@ -54,12 +56,19 @@ public class DoraWorkerClientServiceHandler extends BlockWorkerGrpc.BlockWorkerI
   private final ReadResponseMarshaller mReadResponseMarshaller = new ReadResponseMarshaller();
   private final DoraWorker mWorker;
 
+  private final Cache<String, ListStatusPResponse> mListStatusCache;
+
   /**
    * Creates a new implementation of gRPC BlockWorker interface.
    * @param workerProcess the worker process
    */
   public DoraWorkerClientServiceHandler(WorkerProcess workerProcess) {
     mWorker = workerProcess.getWorker(DoraWorker.class);
+
+    mListStatusCache = CacheBuilder.newBuilder()
+        .maximumSize(1) // Cache only ONE ListStatus result and expires after the following time
+        .expireAfterWrite(Configuration.getDuration(PropertyKey.DORA_UFS_FILE_STATUS_CACHE_TTL))
```

Comment:
We need a separate configuration item for directory listing cache. Dora assumes files are immutable but directories can be updated more frequently.

Reply:
I have the concerns that there have already too many configure keys.
We definitely need to reduce the number of keys in our system.

But let me add another one.